File: C++/Find All Duplicates in an Array.cpp
// https://leetcode.com/problems/find-all-duplicates-in-an-array/


/*
Short Explaination -
IMP : all elements lie in [1,n]
mark the occurring elements by negating the
relevant position.
Now if that position if positive means, it has been negated twice!
*/
class Solution {
public:
	vector<int> findDuplicates(vector<int>& nums) {
		vector<int> ans;
		for (auto x : nums) {
			x = abs(x);
			nums[x - 1] *= -1;
			if (nums[x - 1] > 0)ans.push_back(x);
		}
		return ans;
	}
};

File: Python/Find All Duplicates in an Array.py
# https://leetcode.com/problems/find-all-duplicates-in-an-array/

class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        ans = []
        for item in nums:
            item = abs(item)
            nums[item-1] *= -1
            if(nums[item-1]>0):
                ans.append(item)
        return ans